### DISCUSSION TITLE
Question about disjoint mode

### DISCUSSION THREAD

**Discussion by dawe** [danielegrattarola/spektral#332] [Q&A]
Hello all, I have two datasets with different features and nodes and we wanted to build an autoencoder using spektral with data loaded in disjoint mode. For each dataset I generate a kNN graph. When `spektral` loads the adjancency matrices for datasets A and B, it is done like this:

```
+-+---+
|A|   |
+-+---+
| |   |
| | B |
| |   |
+-+---+
```

and, since both datasets are made of different features (say `f(A)` and `f(B)`), I am passing features padding them to 0 like this:

```
   f(A)  f(B)
  +----+------+
A |abcd|000000|
  +----+------+
  |0000|efghij|
B |0000|klmnop|
  |0000|qrstuv|
  +----+------+
```

When run the decoder, some features that were padded are now non-zero. While I understand why those values appear, I can possibly ignore them, I have some questions:
- Is padding the proper way to pass node features in disjoint mode?
- Should I instead embed my data in another embedding (e.g. PCA) so that the number of features can be the same, even if they are not _aligned_?

**Comment by danielegrattarola** [comment]
Hi, 

the issue here is that Spektral currently does not support heterogeneous features. 
So, padding can be one way of doing that (although I am not sure I've ever seen it in the literature), or alternatively you can use a dedicated linear mapping for each kind of node, to pre-embed all nodes in the same space. 

The correct way to do this would be to have a custom message function for every type of node, directly in the GNN layers, but I have had no time to add new features to the library recently. 

Cheers
